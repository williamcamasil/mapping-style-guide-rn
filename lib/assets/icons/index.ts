export * as Large from './large';
export * as Heavy from './heavy';
export * as Thin from './thin';
export * as Small from './small';
export * as Default from './default';
