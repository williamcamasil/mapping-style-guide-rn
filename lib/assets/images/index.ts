export { default as BtnTakePicture } from './BtnTakePicture';
export { default as BulletPoint } from './BulletPoint';
export { default as CheckboxImg } from './CheckboxImg';
export { default as DialogClose } from './DialogClose/index.svg';
export { default as Lock } from './Lock';
export { default as MaskedCardNumber } from './MaskedCardNumber';
export { default as Pointer } from './Pointer';
export { default as RadioImg } from './RadioImg';
