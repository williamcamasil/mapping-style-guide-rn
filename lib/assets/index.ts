export * as Icons from './icons';
export * as Images from './images';
export * as Lotties from './lotties';
