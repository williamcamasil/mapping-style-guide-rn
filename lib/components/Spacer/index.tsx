import React, { useMemo } from 'react';
import { Text, ViewStyle } from 'react-native';

type SpacerPropsType = {
  size: ViewStyle['height'];
};

const Spacer: React.FC<SpacerPropsType> = ({ size }) => {
  const textStyle = useMemo(() => ({
    height: size,
    width: size,
  }), [size]);
  /*
  * A decisão de usar um Text em vez de um View é para que esse Spacer
  * possa ser usado dentro de outras Views e tbm de outros Text sem restrição.
  */
  return (
    <Text style={textStyle} />
  );
};

export default Spacer;
