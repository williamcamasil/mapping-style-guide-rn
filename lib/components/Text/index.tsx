import React from 'react';
import {
  StyleProp, StyleSheet, Text as RNText, TextProps as RNTextProp, TextStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import {
  ColorPaletesType, FontWeightType, LineHeightType, Typography,
} from '../../tokens';
import {
  getColorStyle, getVariantStyle, getWeightStyle, TextVariantType,
} from './utils';

export type { TextVariantType } from './utils';

export type TextPropTypes = RNTextProp & {
  theme: AppThemeType;
  style?: StyleProp<TextStyle>;
  color?: ColorPaletesType;
  variant?: TextVariantType;
  weight?: FontWeightType;
  lineHeight?: LineHeightType;
};

const styles = StyleSheet.create({
  text: {
    letterSpacing: Typography.letterSpaces.default,
    flexShrink: 1,
  },
});

const Text: React.FC<TextPropTypes> = ({
  children,
  color,
  theme,
  style,
  variant = 'body',
  weight,
  lineHeight,
  allowFontScaling = false,
  ...others
}) => {

  const textStyles = useTextStyles(() => [
    styles.text,
    getColorStyle(theme, color ?? 'neutralGray600'),
    getVariantStyle(theme, variant, lineHeight),
    getWeightStyle(theme, weight),
    style,
  ], [color, lineHeight, style, theme, variant, weight]);

  return (
    <RNText
      {...others}
      allowFontScaling={allowFontScaling}
      style={textStyles}
    >
      {children}
    </RNText>
  );
};

export default withTheme(Text);
