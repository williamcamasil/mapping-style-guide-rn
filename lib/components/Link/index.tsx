import React, { ComponentType } from 'react';
import {
  StyleProp, StyleSheet, TouchableOpacity, TouchableOpacityProps, View, ViewStyle, Text as RNText, TextStyle,
} from 'react-native';

import { SvgProps } from 'react-native-svg';

import { useViewStyles } from '../../hooks/useStyles';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';
import Spacer from '../Spacer';
import Text, { TextPropTypes } from '../Text';
import {
  LinkVariantType, getVariantLinkStyle, getTextColor, getVariantSpaceSize, getIconSize, LinkSizeType, getTextSize,
} from './utils';

export type LinkPropsType<IconProps extends SvgProps = SvgProps> = {
  theme: AppThemeType;
  children: TextPropTypes['children'];
  onPress: TouchableOpacityProps['onPress'];
  testID?: string;
  variant?: LinkVariantType;
  size?: LinkSizeType;
  style?: StyleProp<ViewStyle | TextStyle>;
  disabled?: boolean;
  Icon?: ComponentType<IconProps>;
  textColor?: ColorPaletesType;
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  containerIcon: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

const Link: React.FC<LinkPropsType> = ({
  testID,
  children,
  style,
  onPress,
  disabled,
  theme,
  variant,
  Icon,
  size,
  textColor,
}) => {
  const textVariant = getTextSize(variant, size);

  const textColorName = getTextColor(textColor, variant, disabled, size);

  const buttonStyles = useViewStyles(() => [
    getVariantLinkStyle(theme, textColor, variant, disabled),
    style,
  ], [theme, textColor, variant, disabled, style]);

  const renderText = () => (
    <Text testID="link-text" color={textColorName} variant={textVariant} lineHeight="large">
      {children}
    </Text>
  );

  const renderComponent = () => {
    if (Icon) {
      const spaceSize = getVariantSpaceSize(theme, size);
      const iconSize = getIconSize(theme, size);

      return (
        <View style={styles.containerIcon}>
          <Icon width={iconSize} height={iconSize} color={theme.colors[textColorName]} />
          <Spacer size={spaceSize} />
          {renderText()}
        </View>
      );
    }

    return (
      renderText()
    );
  };

  if (variant === 'text') {
    return (
      <RNText
        testID={testID}
        style={buttonStyles}
        onPress={onPress}
        disabled={disabled}
      >
        {children}
      </RNText>
    );
  }

  return (
    <View style={styles.container}>
      <TouchableOpacity
        testID={testID}
        onPress={onPress}
        disabled={disabled}
        activeOpacity={theme.opacities.intense}
        style={buttonStyles}
      >
        {renderComponent()}
      </TouchableOpacity>
    </View>
  );
};

Link.defaultProps = {
  variant: 'highlightedPrimary',
  size: 'large',
};

export default withTheme(Link);
