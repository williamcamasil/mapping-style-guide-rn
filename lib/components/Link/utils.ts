import { TextStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType, SpacingValueType } from '../../tokens';
import { TextVariantType } from '../Text';

export type LinkVariantType =
  'highlightedPrimary'
  | 'highlightedSecondary'
  | 'primary'
  | 'secondary'
  | 'text';

export type LinkSizeType = 'large' | 'small';

export const getTextColor = (
  textColor?: ColorPaletesType,
  variant?: LinkVariantType,
  disabled?: boolean,
  size?: LinkSizeType,
): ColorPaletesType => {
  if (disabled) return 'neutralGray300';
  if (textColor) return textColor;

  const isSmall = size === 'small';

  switch (variant) {
    case 'secondary':
    case 'highlightedSecondary':
      return isSmall ? 'neutralGray500' : 'neutralGray600';
    case 'highlightedPrimary':
    case 'primary':
    default:
      return isSmall ? 'primaryMain' : 'primary700';
  }
};

export function getTextSize(variant?: LinkVariantType, size?: LinkSizeType): TextVariantType {
  if (variant === 'highlightedPrimary' || variant === 'highlightedSecondary' || size === 'small') {
    return 'small';
  }

  return 'body';
}

export function getVariantSpaceSize(theme: AppThemeType, size?: LinkSizeType): SpacingValueType {
  switch (size) {
    case 'small':
      return theme.spacings.sQuark;
    case 'large':
    default:
      return theme.spacings.sNano;
  }
}

export function getIconSize(theme: AppThemeType, size?: LinkSizeType): SpacingValueType {
  switch (size) {
    case 'small':
      return theme.spacings.sXS;
    case 'large':
    default:
      return theme.spacings.sMedium;
  }
}

function getBorderColor(
  theme: AppThemeType,
  variant: LinkVariantType,
  textColor?: ColorPaletesType,
  disabled?: boolean,
) {
  if (disabled) return theme.colors.neutralGray300;
  if (textColor) return theme.colors[textColor];

  switch (variant) {
    case 'highlightedPrimary':
      return theme.colors.primary700;
    case 'highlightedSecondary':
    default:
      return theme.colors.neutralGray500;
  }
}

export function getVariantLinkStyle(
  theme: AppThemeType,
  textColor?: ColorPaletesType,
  variant?: LinkVariantType,
  disabled?: boolean,
): TextStyle | null {
  switch (variant) {
    case 'text':
      return {
        textDecorationLine: 'underline',
        color: theme.colors[getTextColor(textColor, variant, disabled)],
      };
    case 'highlightedPrimary':
    case 'highlightedSecondary':
      return {
        borderBottomWidth: theme.borders.width.thin,
        paddingBottom: theme.spacings.sNano,
        borderColor: getBorderColor(theme, variant, textColor, disabled),
      };
    default:
      return null;
  }
}
