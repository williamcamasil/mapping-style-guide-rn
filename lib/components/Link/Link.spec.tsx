import React from 'react';

import {
  fireEvent, render, within,
} from '@testing-library/react-native';

import Link from '.';
import { Icons } from '../../assets';
import { DefaultTheme } from '../../theme';
import Spacer from '../Spacer';
import Text from '../Text';

describe('link snapshot', () => {
  it('default', () => {
    const tree = render(
      <Link testID="touchable-link" onPress={() => null} variant="highlightedPrimary" size="small">
        Click me
      </Link>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('disabled', () => {
    const tree = render(
      <Link testID="touchable-link" disabled onPress={() => null} variant="highlightedPrimary" size="small">
        Click me
      </Link>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot of the text variant', () => {
    const tree = render(
      <Link testID="touchable-link" onPress={() => null} variant="text">
        Click me
      </Link>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot of the disabled text variant', () => {
    const tree = render(
      <Link testID="touchable-link" onPress={() => null} variant="text" disabled>
        Click me
      </Link>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot of the text variant with changed color', () => {
    const tree = render(
      <Link testID="touchable-link" onPress={() => null} variant="text" textColor="secondaryMain">
        Click me
      </Link>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot of the text variant with changed size', () => {
    const tree = render(
      <Link testID="touchable-link" onPress={() => null} variant="text" style={{ fontSize: 14 }}>
        Click me
      </Link>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('link', () => {
  it('renders correctly', () => {
    render(
      <Link testID="touchable-link" onPress={() => null} variant="highlightedPrimary" size="small">
        Click me
      </Link>,
    );
  });

  it('testing link action', () => {
    const mockFunction = jest.fn();
    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="highlightedPrimary"
        size="small"
      >
        Click me
      </Link>,
    );

    const primaryLinkInstance = tree.getByTestId('touchable-link');

    fireEvent.press(primaryLinkInstance);

    expect(mockFunction).toHaveBeenCalled();
  });

  it('testing link action disabled', () => {
    const mockFunction = jest.fn();
    const tree = render(
      <Link
        testID="touchable-link"
        disabled
        onPress={(mockFunction)}
        variant="highlightedPrimary"
        size="small"
      >
        Click me
      </Link>,
    );

    const primaryLinkInstance = tree.getByTestId('touchable-link');

    fireEvent.press(primaryLinkInstance);

    expect(mockFunction).not.toHaveBeenCalled();
  });

  it('testing link disabled', () => {
    const tree = render(
      <Link
        testID="touchable-link"
        disabled
        onPress={() => {}}
        variant="highlightedPrimary"
        size="small"
      >
        Click me
      </Link>,
    );
    const primaryLinkInstance = tree.getByTestId('touchable-link').props;
    expect(primaryLinkInstance.accessibilityState).toHaveProperty('disabled', true);
  });

  it('testing variant highlightedPrimary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="highlightedPrimary"
        size="large"
      >
        Click me
      </Link>,
    );

    const textComponent = within(tree.getByTestId('touchable-link')).UNSAFE_getByType(Text);

    expect(textComponent.props).toHaveProperty('color', 'primary700');
    expect(textComponent.props).toHaveProperty('variant', 'small');
  });

  it('testing variant highlightedSecondary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="highlightedSecondary"
        size="small"
      >
        Click me
      </Link>,
    );

    const textComponent = within(tree.getByTestId('touchable-link')).UNSAFE_getByType(Text);

    expect(textComponent.props).toHaveProperty('color', 'neutralGray500');
    expect(textComponent.props).toHaveProperty('variant', 'small');
  });

  it('testing variant primary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="primary"
        size="large"
      >
        Click me
      </Link>,
    );

    const textComponent = within(tree.getByTestId('touchable-link')).UNSAFE_getByType(Text);

    expect(textComponent.props).toHaveProperty('color', 'primary700');
    expect(textComponent.props).toHaveProperty('variant', 'body');
  });

  it('testing variant secondary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="secondary"
        size="large"
      >
        Click me
      </Link>,
    );

    const textComponent = within(tree.getByTestId('touchable-link')).UNSAFE_getByType(Text);

    expect(textComponent.props).toHaveProperty('color', 'neutralGray600');
    expect(textComponent.props).toHaveProperty('variant', 'body');
  });

  it('testing variant primary e test small', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="primary"
        size="small"
      >
        Click me
      </Link>,
    );

    const textComponent = within(tree.getByTestId('touchable-link')).UNSAFE_getByType(Text);

    expect(textComponent.props).toHaveProperty('color', 'primaryMain');
    expect(textComponent.props).toHaveProperty('variant', 'small');
  });

  it('testing variant smallSecondary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="secondary"
        size="small"
      >
        Click me
      </Link>,
    );

    const textComponent = within(tree.getByTestId('touchable-link')).UNSAFE_getByType(Text);

    expect(textComponent.props).toHaveProperty('color', 'neutralGray500');
    expect(textComponent.props).toHaveProperty('variant', 'small');
  });

  it('testing with icon component', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="highlightedPrimary"
        size="small"
        Icon={Icons.Thin.Key}
      >
        Click me
      </Link>,
    );

    const primaryLinkInstance = within(tree.getByTestId('touchable-link'));

    const iconComponent = primaryLinkInstance.UNSAFE_getByType(Icons.Thin.Key);
    const textComponent = primaryLinkInstance.UNSAFE_getByType(Text);
    const spaceComponent = primaryLinkInstance.UNSAFE_getByType(Spacer);

    expect(iconComponent).toBeDefined();
    expect(textComponent).toBeDefined();
    expect(spaceComponent).toBeDefined();
  });

  it('testing with icon component and variant primary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="primary"
        size="large"
        Icon={Icons.Thin.Key}
      >
        Click me
      </Link>,
    );

    const primaryLinkInstance = within(tree.getByTestId('touchable-link'));

    const iconComponent = primaryLinkInstance.UNSAFE_getByType(Icons.Thin.Key);
    const textComponent = primaryLinkInstance.UNSAFE_getByType(Text);
    const spaceComponent = primaryLinkInstance.UNSAFE_getByType(Spacer);

    expect(iconComponent).toBeDefined();
    expect(textComponent).toBeDefined();
    expect(spaceComponent).toBeDefined();
    expect(iconComponent.props).toHaveProperty('width', DefaultTheme.spacings.sMedium);
    expect(iconComponent.props).toHaveProperty('height', DefaultTheme.spacings.sMedium);
    expect(spaceComponent.props).toHaveProperty('size', DefaultTheme.spacings.sNano);
  });

  it('testing with icon component and variant secondary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="secondary"
        size="large"
        Icon={Icons.Thin.Key}
      >
        Click me
      </Link>,
    );

    const primaryLinkInstance = within(tree.getByTestId('touchable-link'));

    const iconComponent = primaryLinkInstance.UNSAFE_getByType(Icons.Thin.Key);
    const textComponent = primaryLinkInstance.UNSAFE_getByType(Text);
    const spaceComponent = primaryLinkInstance.UNSAFE_getByType(Spacer);

    expect(iconComponent).toBeDefined();
    expect(textComponent).toBeDefined();
    expect(spaceComponent).toBeDefined();
    expect(iconComponent.props).toHaveProperty('width', DefaultTheme.spacings.sMedium);
    expect(iconComponent.props).toHaveProperty('height', DefaultTheme.spacings.sMedium);
    expect(spaceComponent.props).toHaveProperty('size', DefaultTheme.spacings.sNano);
  });

  it('testing with icon component and variant primary e size small', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="primary"
        size="small"
        Icon={Icons.Thin.Key}
      >
        Click me
      </Link>,
    );

    const primaryLinkInstance = within(tree.getByTestId('touchable-link'));

    const iconComponent = primaryLinkInstance.UNSAFE_getByType(Icons.Thin.Key);
    const textComponent = primaryLinkInstance.UNSAFE_getByType(Text);
    const spaceComponent = primaryLinkInstance.UNSAFE_getByType(Spacer);

    expect(iconComponent).toBeDefined();
    expect(textComponent).toBeDefined();
    expect(spaceComponent).toBeDefined();
    expect(iconComponent.props).toHaveProperty('width', DefaultTheme.spacings.sXS);
    expect(iconComponent.props).toHaveProperty('height', DefaultTheme.spacings.sXS);
    expect(spaceComponent.props).toHaveProperty('size', DefaultTheme.spacings.sQuark);
  });

  it('testing with icon component and variant smallSecondary', () => {
    const mockFunction = jest.fn();

    const tree = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        variant="secondary"
        size="small"
        Icon={Icons.Thin.Key}
      >
        Click me
      </Link>,
    );

    const primaryLinkInstance = within(tree.getByTestId('touchable-link'));

    const iconComponent = primaryLinkInstance.UNSAFE_getByType(Icons.Thin.Key);
    const textComponent = primaryLinkInstance.UNSAFE_getByType(Text);
    const spaceComponent = primaryLinkInstance.UNSAFE_getByType(Spacer);

    expect(iconComponent).toBeDefined();
    expect(textComponent).toBeDefined();
    expect(spaceComponent).toBeDefined();
    expect(iconComponent.props).toHaveProperty('width', DefaultTheme.spacings.sXS);
    expect(iconComponent.props).toHaveProperty('height', DefaultTheme.spacings.sXS);
    expect(spaceComponent.props).toHaveProperty('size', DefaultTheme.spacings.sQuark);
  });

  it('testing Link with custom textColor', () => {
    const mockFunction = jest.fn();
    const result = render(
      <Link
        testID="touchable-link"
        onPress={mockFunction}
        textColor="feedbackSuccess500"
        size="small"
        Icon={Icons.Thin.Key}
      >
        Click me
      </Link>,
    );

    expect(result.getByTestId('touchable-link').props?.style.borderColor).toEqual(DefaultTheme.colors.feedbackSuccess500);
    expect(result.getByTestId('link-text').props?.style.color).toEqual(DefaultTheme.colors.feedbackSuccess500);
    expect(result.toJSON()).toMatchSnapshot();
  });
});
