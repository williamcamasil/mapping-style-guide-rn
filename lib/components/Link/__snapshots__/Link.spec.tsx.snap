// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link snapshot default 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#0F4199",
        "opacity": 1,
        "paddingBottom": 8,
      }
    }
    testID="touchable-link"
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#013188",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 14,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 20,
        }
      }
      testID="link-text"
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`link snapshot disabled 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#E1E2E4",
        "opacity": 1,
        "paddingBottom": 8,
      }
    }
    testID="touchable-link"
  >
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#E1E2E4",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 14,
          "fontWeight": "bold",
          "letterSpacing": -0.02,
          "lineHeight": 20,
        }
      }
      testID="link-text"
    >
      Click me
    </Text>
  </View>
</View>
`;

exports[`link snapshot should match snapshot of the disabled text variant 1`] = `
<Text
  disabled={true}
  onPress={[Function]}
  style={
    {
      "color": "#E1E2E4",
      "textDecorationLine": "underline",
    }
  }
  testID="touchable-link"
>
  Click me
</Text>
`;

exports[`link snapshot should match snapshot of the text variant 1`] = `
<Text
  onPress={[Function]}
  style={
    {
      "color": "#0F4199",
      "textDecorationLine": "underline",
    }
  }
  testID="touchable-link"
>
  Click me
</Text>
`;

exports[`link snapshot should match snapshot of the text variant with changed color 1`] = `
<Text
  onPress={[Function]}
  style={
    {
      "color": "#E95814",
      "textDecorationLine": "underline",
    }
  }
  testID="touchable-link"
>
  Click me
</Text>
`;

exports[`link snapshot should match snapshot of the text variant with changed size 1`] = `
<Text
  onPress={[Function]}
  style={
    {
      "color": "#0F4199",
      "fontSize": 14,
      "textDecorationLine": "underline",
    }
  }
  testID="touchable-link"
>
  Click me
</Text>
`;

exports[`link testing Link with custom textColor 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderBottomWidth": 1,
        "borderColor": "#028E39",
        "opacity": 1,
        "paddingBottom": 8,
      }
    }
    testID="touchable-link"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={16}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGGroup
            clipPath="a"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278357561,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="1.6"
          >
            <RNSVGPath
              d="m16.209 14.035-.992-1.008-1.26 1.24-2.124-.019-.86 2.08a1.999 1.999 0 0 1-1.866 1.236l-4.163-.035A2.001 2.001 0 0 1 3.1 16.262l-1.408-3.575a1.999 1.999 0 0 1 .013-1.498l1.469-3.55a2 2 0 0 1 1.865-1.236l4.163.036a2 2 0 0 1 1.843 1.267l.805 2.04 8.52.073 2.23 2.27-2.269 2.23-1.875-.015-1.24-1.261-1.008.992h0Z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            />
            <RNSVGPath
              d="M6.727 11.464a.433.433 0 1 1-.001.002"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
          <RNSVGDefs>
            <RNSVGClipPath
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              name="a"
            >
              <RNSVGPath
                d="M0 0h24v24H0z"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              />
            </RNSVGClipPath>
          </RNSVGDefs>
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          {
            "height": 4,
            "width": 4,
          }
        }
      />
      <Text
        allowFontScaling={false}
        style={
          {
            "color": "#028E39",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 14,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 20,
          }
        }
        testID="link-text"
      >
        Click me
      </Text>
    </View>
  </View>
</View>
`;
